aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -675,12 +675,13 @@ TensorIterator TensorIterator::comparison_op(Tensor& out, const Tensor& a,
 }
 
 TensorIterator TensorIterator::unary_op(Tensor& out, const Tensor& a,
-    bool check_mem_overlap) {
+    bool check_mem_overlap, bool promoting) {
   auto iter = TensorIterator();
   iter.set_check_mem_overlap(check_mem_overlap);
   iter.add_output(out);
   iter.add_input(a);
   iter.num_outputs_ = 1;
+  if(promoting == true) iter.promote_common_dtype();
```

Comment:
promote_common_dtype implies we're finding the common dtype across all operands and setting the output dtype based on that. These are unary ops, so this doesn't look right.

Also, nit, don't need to check `==true` when using a boolean in a condition.